src/test/java/com/kodilla/ecommercee/repository/GroupRepositoryTestSuite.java

```diff
@@ -0,0 +1,99 @@
+package com.kodilla.ecommercee.repository;
+
+import com.kodilla.ecommercee.domain.Group;
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.service.GroupDbService;
+import com.kodilla.ecommercee.service.ProductDbService;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class GroupRepositoryTestSuite {
+    @Autowired
+    private GroupRepository groupRepository;
+    @Autowired
+    private GroupDbService groupDbService;
+    @Autowired
+    private ProductDbService productDbService;
+
+    private Group getTestGroup() {
+        String name = "dresses";
+        List<Product> products = new ArrayList<>();
+        Group dresses = new Group(null, name, products);
+        Product product1 = new Product(null, "P1", "Pellentesque tempus interdum quam ut rhoncus.", BigDecimal.valueOf(230), dresses);
+        Product product2 = new Product(null, "P2", "Tempus interdum quam ut rhoncus.", BigDecimal.valueOf(255), dresses);
+        Product product3 = new Product(null, "P3", "Interdum quam ut rhoncus.", BigDecimal.valueOf(543), dresses);
+        products.add(product1);
+        products.add(product2);
+        products.add(product3);
+
+        return dresses;
+    }
+
+    @Test
+    public void testSaveGroupWithProducts() {
+        //Given
+        Group group = getTestGroup();
+
+        //When
+        groupRepository.save(group);
```

Comment:
Ten fragment pojawia sie w kazdym tescie, mozna by to do @BeforeClass wrzucic

Reply:
W trzecim teście jest trochę inaczej - pod tworzonym obiektem group pojawia się linijka z tworzonym obiektem newGroupName; Poza tym teraz kod jest czytelny, natomiast jeśli wrzucilibyśmy ten fragment do befora nie byłby już tak czytelny.